jgrapht-core/src/test/java/org/jgrapht/alg/treedecompostion/NiceDecompositionUtil.java

```diff
@@ -0,0 +1,114 @@
+package org.jgrapht.alg.treedecompostion;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.*;
+import java.util.Map.*;
+
+import org.jgrapht.*;
+import org.jgrapht.graph.*;
+
+public class NiceDecompositionUtil
+{
+    private NiceDecompositionUtil() {}
+    
+    /**
+     * tests whether the tree decomposition (decomposition, map) with root is nice
+     * 
+     * @param decomposition the tree decomposition to check
+     * @param map the map of the tree decomposition
+     * @param root the root of the tree decomposition
+     */
+    public static <V,E,W> void testNiceDecomposition(Graph<V,E> decomposition, Map<V,Set<W>> map, V root){
+        
+        Queue<V> queue = new LinkedList<V>();
+        
+        //test and add root
+        assertTrue(root+" is no valid root"
+                + "\n in decomposition "+decomposition
+                + "\n and map"+map, map.get(root).size() == 0);
```

Comment:
use set.isEmpty() instead.

Reply:
Can you reaname the package treedecomposition to decomposition? see https://github.com/jgrapht/jgrapht/issues/586